### PR TITLE
[CWS] Allow suppressing all non-drift rule-based events

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

This PR modify the auto-suppression behavior.

To be suppressed:
* the matching rule has to have the `allow_autosuppression=true` tag.
* the event type should be defined in `runtime_security_config.security_profile.auto_suppression.event_types`
* the event should have an associated containerID
* if the workload have an active dump: we should have the `runtime_security_config.activity_dump.auto_suppression.enabled` config set to true.
* if the workload has an active security profile (either in learning, or in stable with the event already present): we should have the `runtime_security_config.security_profile.auto_suppression.enabled` config set to true.

And now if a workload has no active security profile, and neither an active activity dump (this happens when a new workload starts while we already reached the maximum of concurrent active dump), we can allow events to be suppressed if `runtime_security_config.activity_dump.auto_suppression.enabled` config is set to true.

### Motivation

Events could be suppressed while an activity dump is running, most of the remaining noise comes from workloads that are in wait list to be dumped. This change should allow to remove most of the remaining noise.

### CODE REVIEW

**Review by jhgilbert — APPROVED** [review]

File: pkg/security/tests/module_tester_linux.go [hunk 1]
```diff
@@ -1301,12 +1301,10 @@ func (tm *testModule) StartADocker() (*dockerCmdWrapper, error) {
 func (tm *testModule) GetDumpFromDocker(dockerInstance *dockerCmdWrapper) (*activityDumpIdentifier, error) {
 	dumps, err := tm.ListActivityDumps()
 	if err != nil {
-		_, _ = dockerInstance.stop()
```
Comment (Gui774ume) [inline pkg/security/tests/module_tester_linux.go:1304]: Any particular reason for this ? Could we fail to list activity dumps and yet still have the docker container running ? 🤔

File: pkg/security/tests/activity_dumps_test.go [hunk 2]
```diff
@@ -636,3 +637,113 @@ func TestActivityDumpsAutoSuppression(t *testing.T) {
...
+	var expectedFormats = []string{"profile", "protobuf"}
+	var testActivityDumpTracedEventTypes = []string{"exec", "open", "syscalls", "dns", "bind"}
+
+	outputDir := t.TempDir()
+	os.MkdirAll(outputDir, 0755)
+	defer os.RemoveAll(outputDir)
+
+	rulesDef := []*rules.RuleDefinition{
+		{
+			ID:         "test_autosuppression_exec",
```
Comment (Gui774ume) [inline pkg/security/tests/activity_dumps_test.go:664]: Should we add a test case where `allow_autosuppression: true` and `workload_drift_only: false` and make sure non drift events from workloads that haven't been learnt yet aren't suppressed ? 🤔

File: pkg/security/rules/autosuppression/autosuppression.go [hunk 3]
```diff
@@ -40,60 +49,66 @@ type Opts struct {
...
 // Apply resets the auto suppression stats based on the given ruleset
 func (as *AutoSuppression) Apply(ruleSet *rules.RuleSet) {
+	var enabledSuppressionTypes []string
+	if as.opts.SecurityProfileAutoSuppressionEnabled {
+		enabledSuppressionTypes = append(enabledSuppressionTypes, securityProfileSuppressionType)
+		enabledSuppressionTypes = append(enabledSuppressionTypes, noWorkloadDriftSuppressionType)
+	}
+	if as.opts.ActivityDumpAutoSuppressionEnabled {
+		enabledSuppressionTypes = append(enabledSuppressionTypes, activityDumpSuppressionType)
+	}
```
Comment (Gui774ume) [inline pkg/security/rules/autosuppression/autosuppression.go:103]: I _**think**_ Yoann wanted to stick the "workload drift suppression" to the "activity dump suppression" no (or at least that's what he wrote in the description of the PR) ?

Should we have a third config parameter ? 🤔

Comment (spikat) [reply pkg/security/rules/autosuppression/autosuppression.go:103]: Good catch!
At start it was under the "security profile" config but I moved it to the "activity dump" one. And that's me who wrote the desc :smile: 
Will be fixed on upcoming commit

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**